### PR TITLE
[WXPO] (debugging) Sideloading to web, sideloading for Outlook, and more

### PR DESCRIPTION
**Pull request by mscharlock**
Hi all! I **think** I put updates in the right spots. In order to add images that will be compliant I have to take some new screenshots, but I figure we should at least get info out and then I can add pictures of dialog boxes if needed in a second pull. :)

### CODE REVIEW

**Review by ElizabethSamuel-MSFT — COMMENTED** [review]
Sharing the comments I've made so far. Still reviewing.

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 1]
```diff
@@ -45,24 +74,30 @@ The following video walks you through the process of sideloading your add-in in
...
 
 ## Remove a sideloaded add-in
 
-You can remove a previously sideloaded add-in by clearing your browser's cache. Additionally, if you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to clear the cache and then re-sideload the add-in using updated manifest. Doing so will allow Office to render the add-in as it's described by the updated manifest.
+You can remove a previously sideloaded add-in by clearing your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), [you may need to clear the Office cache](clear-cache.md) and then re-sideload the add-in using updated manifest. Doing so will allow Office to render the add-in as it's described by the updated manifest.
+
+## See also
+
+- [Sideload Office Add-ins on iPad and Mac](sideload-an-office-add-in-on-ipad-and-mac.md)
+- [Sideload Outlook Add-ins for testing](../outlook/sideload-outlook-add-ins-for-testing.md)
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/sideload-office-add-ins-for-testing.md:102]:
```suggestion
- [Sideload Outlook add-ins for testing](../outlook/sideload-outlook-add-ins-for-testing.md)
```

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 2]
```diff
@@ -45,24 +74,30 @@ The following video walks you through the process of sideloading your add-in in
...
+7. Right click the web project and choose **Debug** > **Start new instance**. This will run the web project without launching Office.
 8. From Office on the web, sideload the add-in using steps previously described in [Sideload an Office Add-in in Office on the web](#sideload-an-office-add-in-in-office-on-the-web).
 
 ## Remove a sideloaded add-in
 
-You can remove a previously sideloaded add-in by clearing your browser's cache. Additionally, if you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to clear the cache and then re-sideload the add-in using updated manifest. Doing so will allow Office to render the add-in as it's described by the updated manifest.
+You can remove a previously sideloaded add-in by clearing your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), [you may need to clear the Office cache](clear-cache.md) and then re-sideload the add-in using updated manifest. Doing so will allow Office to render the add-in as it's described by the updated manifest.
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/sideload-office-add-ins-for-testing.md:97]:
```suggestion
You can remove a previously sideloaded add-in by clearing your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to [clear the Office cache](clear-cache.md) and then re-sideload the add-in using the updated manifest. Doing so will allow Office to render the add-in as it's described by the updated manifest.
```

File: docs/outlook/autolaunch.md [hunk 3]
```diff
@@ -194,10 +194,10 @@ In this scenario, you'll add handling for composing new items.
 
 ## Try it out
 
-1. Run the following command in the root directory of your project. When you run this command, the local web server will start (if it's not already running).
+1. Run the following command in the root directory of your project. When you run this command, the local web server will start (if it's not already running) and your add-in will sideload.
```
[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/outlook/autolaunch.md:197]: Yeah I struggled with whether to use to be sideloaded or just sideload. Also, sideload = the most annoying verb ever

**Comment by mscharlock** [comment]
I should mention that I took MZ's video out of this article, because it is from way back in the day and references adding your add-ing to the catalog, which I am pretty sure nobody has to do anymore. I left that in the text but if that is true (that it is no longer a thing) then we need to take it out of docs in a couple of places.

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 4]
```diff
@@ -1,30 +1,59 @@
...
-The following video walks you through the process of sideloading your add-in in Office on the web or desktop.
+This process is supported for **Excel**, **OneNote**, **PowerPoint**, and **Word** only. For other host applications, see [the manual sideloading instructions following this section](sideload-office-add-ins-for-testing.md#sideload-an-office-add-in-in-office-on-the-web-manually). The below project assumes that you are using a project created with [the Yeoman generator for Office Add-ins](https://github.com/OfficeDev/generator-office).
 
-> [!VIDEO https://www.youtube.com/embed/XXsAw2UUiQo]
+1. Open [Office on the web](https://office.live.com/). Using the **Create** option, make a document in **Excel**, **OneNote**, **PowerPoint**, or **Word**. In this new document, select **Share** in the ribbon, select **Copy Link**, and copy the URL.
 
-## Sideload an Office Add-in in Office on the web
+2. In the root directory of your yo office project files, open the **package.json** file. Within the **scripts** section of this file, create a **"document"** property. Paste the URL you copied in as the value for the "document" property. For example, yours will look something like this:
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/sideload-office-add-ins-for-testing.md:25]:
```suggestion
2. In the root directory of your yo office project files, open the **package.json** file. Within the **scripts** section of this file, create a `"document"` property. Paste the URL you copied as the value for the "document" property. For example, yours will look something like this:
```

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 5]
```diff
@@ -1,30 +1,59 @@
...
+    ...
+  }
+```
+
+>[!TIP]
+>If you are creating an add-in not using our Yeoman generator, you can add query parameters to your document's URL, by appending the following to the existing URL:
+
+- The dev server port, such as `&wdaddindevserverport=3000`.
+- The manifest file name, such as `&wdaddinmanifestfile=manifest1.xml`.
+- The manifest GUID, such as `&wdaddinmanifestguid=05c2e1c9-3e1d-406e-9a91-e9ac64854143`.
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/sideload-office-add-ins-for-testing.md:39]:
```suggestion
  > [!TIP]
  > If you are creating an add-in not using our Yeoman generator, you can add query parameters to your document's URL, by appending the following to the existing URL:

    - The dev server port, such as `&wdaddindevserverport=3000`.
    - The manifest file name, such as `&wdaddinmanifestfile=manifest1.xml`.
    - The manifest GUID, such as `&wdaddinmanifestguid=05c2e1c9-3e1d-406e-9a91-e9ac64854143`.
```

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 6]
```diff
@@ -1,30 +1,59 @@
...
 
-## Sideload an Office Add-in in Office on the web
+2. In the root directory of your yo office project files, open the **package.json** file. Within the **scripts** section of this file, create a **"document"** property. Paste the URL you copied in as the value for the "document" property. For example, yours will look something like this:
+
+```json
+  "scripts": {
+    "document": "<YOUR URL>",
+    ...
+  }
+```
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/sideload-office-add-ins-for-testing.md:32]: You may need to play around with the indenting of this and the following tip to ensure they display properly and the numbering will work correctly
```suggestion
  ```json
    "scripts": {
      "document": "<YOUR URL>",
      ...
    }
  ```
```

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 7]
```diff
@@ -1,30 +1,59 @@
...
+>[!TIP]
+>If you are creating an add-in not using our Yeoman generator, you can add query parameters to your document's URL, by appending the following to the existing URL:
+
+- The dev server port, such as `&wdaddindevserverport=3000`.
+- The manifest file name, such as `&wdaddinmanifestfile=manifest1.xml`.
+- The manifest GUID, such as `&wdaddinmanifestguid=05c2e1c9-3e1d-406e-9a91-e9ac64854143`.
 
-1. Open [Office on the web](https://office.live.com/).
+> If you are using the Yeoman generator, adding this information is not necessary as the Yeoman tooling appends this information automatically.
+> Note that in both cases, however, you can only load manifests from localhost.
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/sideload-office-add-ins-for-testing.md:42]:
```suggestion
  > If you are using the Yeoman generator, adding this information is not necessary as the Yeoman tooling appends this information automatically.
  > Note that in both cases, however, you can only load manifests from localhost.
```

**Comment by ElizabethSamuel-MSFT** [comment]
> I should mention that I took MZ's video out of this article, because it is from way back in the day and references adding your add-ing to the catalog, which I am pretty sure nobody has to do anymore. I left that in the text but if that is true (that it is no longer a thing) then we need to take it out of docs in a couple of places.

@mscharlock Which article are you referring to with this comment?

I think the [catalog](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/create-a-network-shared-folder-catalog-for-task-pane-and-content-add-ins) is still available though. However, can you confirm that we don't need to call it out? Then we can certainly remove references to catalog in a couple of these articles.

**Comment by mscharlock** [comment]
> @mscharlock Which article are you referring to with this comment?
> 

Would be the article about overall sideloading: https://docs.microsoft.com/en-us/office/dev/add-ins/testing/sideload-office-add-ins-for-testing. Bc if you can now use the command line to sideload and use yo office templates that just do this for you, I think it makes sense to get rid of it. Esp since it is at the top of the page, which to me sort of implies this is the best way to do the thing

**Comment by TCourtneyOwen** [comment]
Our UI dialog for sideload to web contains a link to https://docs.microsoft.com/en-us/office/dev/add-ins/testing/sideload-office-add-ins-for-testing#:~:text=%20Sideload%20an%20Office%20Add-in%20in%20Office%20on,Add-ins%2C%20and%20then%20Upload%20My%20Add-in.%20More%20.  

I just want to be sure these changes will be accessible from that link.  Can you please clarify if that link will take the user to these changes?

image

**Comment by ElizabethSamuel-MSFT** [comment]
> Our UI dialog for sideload to web contains a link to https://docs.microsoft.com/en-us/office/dev/add-ins/testing/sideload-office-add-ins-for-testing#:~:text=%20Sideload%20an%20Office%20Add-in%20in%20Office%20on,Add-ins%2C%20and%20then%20Upload%20My%20Add-in.%20More%20.
> 
> I just want to be sure these changes will be accessible from that link. Can you please clarify if that link will take the user to these changes?
> 
> image

@TCourtneyOwen Can the text in the dialog say "Office on the web" instead of "Office Online"?

**Comment by TCourtneyOwen** [comment]
@ElizabethSamuel-MSFT This would be a very expensive change unfortunately.  I would have to make the change in the various host tenants and we are talking hours of work to make the change

**Comment by ElizabethSamuel-MSFT** [comment]
> @ElizabethSamuel-MSFT This would be a very expensive change unfortunately. I would have to make the change in the various host tenants and we are talking hours of work to make the change

Thanks, @TCourtneyOwen. I think it should be put into the backlog though since Office Online is no longer a recommended term/brand.

**Comment by ElizabethSamuel-MSFT** [comment]
> Our UI dialog for sideload to web contains a link to https://docs.microsoft.com/en-us/office/dev/add-ins/testing/sideload-office-add-ins-for-testing#:~:text=%20Sideload%20an%20Office%20Add-in%20in%20Office%20on,Add-ins%2C%20and%20then%20Upload%20My%20Add-in.%20More%20.
> 
> I just want to be sure these changes will be accessible from that link. Can you please clarify if that link will take the user to these changes?
> 
> image

The URL https://docs.microsoft.com/en-us/office/dev/add-ins/testing/sideload-office-add-ins-for-testing is the public, live link for this article. Once the changes in the PR are merged then published, the updates should show up.

**Comment by TCourtneyOwen** [comment]
@ElizabethSamuel-MSFT It's a quick change for Word and OneNote as they work in the MonoRepo, but more time consuming for Excel and PPT, as I would need to set up tenants for both.  I can make the change for Word and OneNote today perhaps and then the others can trickle in after.

**Comment by TCourtneyOwen** [comment]
I was wrong - I can make this string change in the MonoRepo and it will propagate to all hosts, so this is an easy change to update the dialog string to "Office on the web"

**Review by TCourtneyOwen — APPROVED** [review]

**Comment by TCourtneyOwen** [comment]
FYI, I am checking in the change to update the Developer Mode opt-in dialog to look like this:

image

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 8]
```diff
@@ -1,30 +1,59 @@
...
 
 # Sideload Office Add-ins in Office on the web for testing
 
-You can install an Office Add-in for testing without having to first put it in an add-in catalog by using sideloading. Sideloading can be done in either Microsoft 365 or Office on the web. The procedure is slightly different for the two platforms.
+When you sideload an add-in, you're able to install the add-in without first putting it in the add-in catalog. This is useful when testing and developing your add-in because you can see how your add-in will appear and function.
 
-When you sideload an add-in, the add-in manifest is stored in the browser's local storage, so if you clear the browser's cache, or switch to a different browser, you have to sideload the add-in again.
+When you sideload an add-in, the add-in's manifest is stored in the browser's local storage, so if you clear the browser's cache, or switch to a different browser, you have to sideload the add-in again.
+
+Sideloading can be done in either Microsoft 365 or Office on the web. The procedure is slightly different for the two platforms.
```
[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/testing/sideload-office-add-ins-for-testing.md:14]: Good point. I suppose that we are comparing desktop vs web although technically with this Outlook update it also would do it on mobile. So...hrm. Let me think about this one.

Comment (mscharlock) [inline docs/testing/sideload-office-add-ins-for-testing.md:14]:
```suggestion
Sideloading can vary between host applications. 
```

File: docs/testing/sideload-office-add-ins-for-testing.md [hunk 9]
```diff
@@ -1,30 +1,59 @@
...
 
 > [!NOTE]
-> Sideloading as described in this article is supported on Word, Excel, and PowerPoint. To sideload an Outlook add-in, see [Sideload Outlook add-ins for testing](../outlook/sideload-outlook-add-ins-for-testing.md).
+> Sideloading as described in this article is supported on Excel, OneNote, PowerPoint and Word. To sideload an Outlook add-in, see [Sideload Outlook add-ins for testing](../outlook/sideload-outlook-add-ins-for-testing.md).
+
+## Sideload an Office Add-in in Office on the web
 
-The following video walks you through the process of sideloading your add-in in Office on the web or desktop.
+This process is supported for **Excel**, **OneNote**, **PowerPoint**, and **Word** only. For other host applications, see [the manual sideloading instructions following this section](sideload-office-add-ins-for-testing.md#sideload-an-office-add-in-in-office-on-the-web-manually). The below project assumes that you are using a project created with [the Yeoman generator for Office Add-ins](https://github.com/OfficeDev/generator-office).
```
[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/testing/sideload-office-add-ins-for-testing.md:21]: This is one I always forget! I'm out of practice, haha

File: docs/testing/sideload-office-add-ins-for-testing.md (hunk 6 above)
Comment (mscharlock) [reply docs/testing/sideload-office-add-ins-for-testing.md:32]: Ok good note - I will accept and then check

**Pull request merged**